### PR TITLE
[config] Changed 'ConfigDescriptionParameterDTO' field serialization 'defaultValue' -> 'default'

### PR DESCRIPTION
**Pull request by cweitkamp**
- Changed `ConfigDescriptionParameterDTO` field serialization: "defaultValue" -> "default"
- Fixed a minor bug in `ConfigDescriptionDTOMapper` for field "unit"
- Added unit tests for `ConfigDescriptionDTOMapper`, `ConfigDescriptionResource` and `AutomationIntegrationJsonTest`

Supersedes #2376
Related to #2148

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/dto/ConfigDescriptionParameterDTO.java [hunk 1]
```diff
@@ -29,6 +30,7 @@
 public class ConfigDescriptionParameterDTO {
 
     public String context;
+    @SerializedName(value = "default")
```
Comment (cweitkamp) [inline bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/dto/ConfigDescriptionParameterDTO.java:33]: As mentioned in https://github.com/openhab/openhab-core/pull/2376#discussion_r637870924 we can use the following syntax to allow both names for parsing.

```suggestion
    @SerializedName(value="default", alternate="defaultValue")
```

Comment (kaikreuzer) [reply bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/dto/ConfigDescriptionParameterDTO.java:33]: Sounds like the safer option in any case. Shall we use this one? What is your preference?

Comment (cweitkamp) [reply bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/dto/ConfigDescriptionParameterDTO.java:33]: I'd prefer adding the fallback.

**Review by kaikreuzer — APPROVED** [review]
Lgtm, thank you!
I'll wait with merging it until after the 3.1.0.M5 build - just to make sure that we do not introduce any regression that we might have overlooked.

**Pull request merged**